contracts/ERC721Collateralizer.sol

```diff
@@ -0,0 +1,426 @@
+
+        // Determine debtor of the collateral.
+        address debtor = agreementToDebtor[agreementId];
+
+        // Mark agreement's collateral as withdrawn by setting the agreement's debtor to 0x0.
+        delete agreementToDebtor[agreementId];
+
+        // Ensure that the debt is not in a state of default.
+        require(
+            termsContract.getExpectedRepaymentValue(
+                agreementId,
+                termsContract.getTermEndTimestamp(agreementId)
+            ) <= termsContract.getValueRepaidToDate(agreementId)
+        );
+
+        // Instantiate an instance of the token, in order to initiate a transfer.
+        ERC721 erc721token = ERC721(collateralTokenAddress);
+
+        // Apply a hack for CryptoKitties, granting transfer approval to this contract itself.
+        if (cryptoKittiesContract == collateralTokenAddress) {
+            erc721token.approve(address(this), collateralTokenID);
+        }
+
+        // Transfer the collateral this contract was holding in escrow back to the debtor.
+        erc721token.transferFrom(address(this), debtor, collateralTokenID);
+
+        // Ensure that transfer has succeeded.
+        require(erc721token.ownerOf(collateralTokenID) == debtor);
+
+        // Log the return event.
+        CollateralReturned(
+            agreementId,
+            debtor,
+            collateralTokenAddress,
+            collateralTokenID
+        );
+    }
+
+    /**
+     * Seizes the collateral from the given debt agreement and
+     * transfers it to the debt agreement's current beneficiary
+     * (i.e. the person who "owns" the debt).
+     *
+     * @param agreementId bytes32 The debt agreement's ID
+     */
+    function seizeCollateral(
+        bytes32 agreementId
+    )
+        public
+        whenNotPaused
+    {
+
+        // The address of the ERC721 contract.
+        address collateralTokenAddress;
+        // The token id, associating the collateral token and the given ERC721 contract.
+        uint256 collateralTokenID;
+        // The terms contract according to which this asset is being collateralized.
+        TermsContract termsContract;
+
+        // Fetch all relevant collateralization parameters.
+        (
+            collateralTokenAddress,
+            collateralTokenID,
+            termsContract
+        ) = retrieveCollateralParameters(agreementId);
+
+        // Seizure can only occur if the collateral has yet to be withdrawn.
+        // When we withdraw collateral, we reset the collateral agreement
+        // in a gas-efficient manner by resetting the address of the debtor to 0.
+        require(agreementToDebtor[agreementId] != address(0));
+
+        // Mark agreement's collateral as withdrawn by setting the agreement's debtor to 0x0.
+        delete agreementToDebtor[agreementId];
+
+        // Ensure debt is in a state of default.
+        require(
+            termsContract.getExpectedRepaymentValue(
```

Comment:
This would be out of scope for this PR but feels like the terms contract should be able to tell you that it's in a state of default.